### PR TITLE
[#7] [Integrate] As a user, I can log in to the app

### PR DESCRIPTION
**Pull request by luongvo**
https://github.com/luongvo/flutter-survey/issues/7

## What happened 👀

When the user comes to the Login screen, they can log in using their email and password.
 
## Insight 📝

- Disable Login button while executing the API
- Validate email and password
- If successful, navigate the user to the Home screen
- If failed, show the error message
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/133330421-c3ad9a27-6d88-4b97-a345-fc246ae1fe36.mp4

### CODE REVIEW

File: lib/pages/login/login_page.dart [hunk 1]
```diff
@@ -129,34 +193,36 @@ class LoginPage extends StatelessWidget {
...
+    }
+    return null;
+  }
+
+  String? _passwordValidator(String? value) {
+    if (value == null || value.length < PASSWORD_LENGTH_MIN) {
+      return AppLocalizations.of(context)!.validationErrorEmailInvalid;
+    }
+    return null;
+  }
```
Comment (markgravity) [inline lib/pages/login/login_page.dart:215]: Should these logic move to LoginModel? 🤔

Comment (luongvo) [reply lib/pages/login/login_page.dart:215]: @markgravity actually they are ui interaction, we try to return the error back to the textfield on typing 💭 So imo, it's better to keep them here

Comment (luongvo) [reply lib/pages/login/login_page.dart:215]: @markgravity check [this](https://github.com/luongvo/flutter-survey/pull/51/files#diff-33389b53b7ec6b6718dcbc20e01367094076fec4fd39356ac5afcced83972816R151) 🤔

File: lib/pages/login/login_model.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/pages/login/login_state.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_survey/usecase/login_use_case.dart';
+
+class LoginModel extends StateNotifier<LoginState> {
```
Comment (sleepylee) [inline lib/pages/login/login_model.dart:7]: Model sounds like a normal `Model` class in any architecture: M(odel)VVM, M(odel)VP...🤔  shouldn't you consider `LoginViewModel` like we always do? 🤷‍♂️

File: lib/pages/login/login_page.dart [hunk 3]
```diff
@@ -129,34 +193,36 @@ class LoginPage extends StatelessWidget {
...
+
+  Future<void> _attemptLogin() async {
+    if (_formKey.currentState!.validate()) {
+      KeyboardUtil.hideKeyboard(context);
+
+      final LoginModel loginModel =
+          context.read<LoginModel>(loginModelProvider.notifier);
+      loginModel.login(_emailController.text, _passwordController.text);
+    } else {
+      return null;
```
Comment (sleepylee) [inline lib/pages/login/login_page.dart:225]: do we really need to return here? 🤔  even if we do, I don't think we need the `else` as the return type is always expected as void

Comment (luongvo) [reply lib/pages/login/login_page.dart:225]: dart's mystery lol, thanks

File: lib/utils/keyboard_util.dart [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import 'package:flutter/cupertino.dart';
+
+class KeyboardUtil {
+  KeyboardUtil._();
```
Comment (sleepylee) [inline lib/utils/keyboard_util.dart:4]: not sure yet how is it in flutter when writing tests, but using static `util` class like this oftentimes caused us trouble in native 🤔

File: lib/pages/login/login_model.dart (hunk 2 above)
Comment (luongvo) [reply lib/pages/login/login_model.dart]: Fixed https://github.com/luongvo/flutter-survey/pull/51/commits/5d4d5216ce5dc0496ab1af80707465e19f95e0c6

File: lib/utils/keyboard_util.dart (hunk 4 above)
Comment (luongvo) [reply lib/utils/keyboard_util.dart:4]: Okay, I will take a look at it later in UI tests 👌

File: lib/pages/login/login_page.dart (hunk 3 above)
Comment (luongvo) [reply lib/pages/login/login_page.dart]: Fixed https://github.com/luongvo/flutter-survey/pull/51/commits/9f123f22f896223a90f5fb92a347386bd6d970fc

**Review by markgravity — APPROVED** [review]
LGTM

**Review by manh-t — APPROVED** [review]

**Pull request merged**